pkg/controller/volume/persistentvolume/pv_controller.go

```diff
@@ -571,6 +571,17 @@ func (ctrl *PersistentVolumeController) syncVolume(volume *v1.PersistentVolume)
 			}
 			return nil
 		} else if claim.Spec.VolumeName == "" {
+			if isMisMatch, err := checkVolumeModeMisMatches(&claim.Spec, &volume.Spec); err != nil || isMisMatch {
+				// Binding for the volume won't be called in syncUnboundClaim,
+				// because findBestMatchForClaim won't return the volume due to volumeMode mismatch.
+				volumeMsg := fmt.Sprintf("Cannot bind PersistentVolume to requested PersistentVolumeClaim %s due to incompatible volumeMode.", claim.Name)
```

Comment:
I prefer %q so that the name is quoted

Reply:
Can you also add some unit tests to binder_test.go?  It should already support checking for events.

Reply:
I changed the code as suggested and added 4 unit tests for syncVolume. Please review v7 patch.